src/ChargeDrift/ChargeDrift.jl

```diff
@@ -196,10 +196,15 @@ function _check_and_update_position!(
             crossing_pos::CartesianPoint{T}, cd_point_type::UInt8, surface_normal::CartesianVector{T} = 
                 get_crossing_pos(det, point_types, copy(current_pos[n]), current_pos[n] + step_vectors[n])
             if cd_point_type == CD_ELECTRODE
-                done[n] = true
-                drift_path[n,istep] = crossing_pos
-                current_pos[n] = crossing_pos
-            elseif cd_point_type == CD_FLOATING_BOUNDARY
+                if crossing_pos in det.contacts
+                    done[n] = true
+                    drift_path[n,istep] = crossing_pos
+                    current_pos[n] = crossing_pos
+                else
+                    cd_point_type = CD_FLOATING_BOUNDARY
+                end

```

Comment:
Something like `geometry_check` in line 199 that would be passed down to this function and then allow/disable the geometry check that you are proposing in this PR.
```suggestion
                if !geometry_check || crossing_pos in det.contacts
                    done[n] = true
                    drift_path[n,istep] = crossing_pos
                    current_pos[n] = crossing_pos
                else
                    cd_point_type = CD_FLOATING_BOUNDARY
                end
```